src/mame/drivers/dec8.cpp

```diff
@@ -1383,6 +1330,10 @@ static INPUT_PORTS_START( csilver )
 	PORT_DIPSETTING(    0x40, DEF_STR( Off ) )
 	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
 	PORT_DIPUNUSED( 0x80, IP_ACTIVE_LOW )
+
+	PORT_START("I8751") /* hooked up on the i8751 */
```

Comment:
This definition is clearly incomplete. This port also contains the interrupt inputs (falling edge or active low).

Reply:
dink said there was no interrupt handler on the MCU for it

Reply:
INT0 is completely blank in this mcu (ref: ida)

code:00000003                         ; External interrupt 0
code:00000003
code:00000003                                         ; public IE0
code:00000003                         IE0:                                    ; CODE XREF: IE0↓j
code:00000003 80 FE                                   sjmp    IE0             ; External interrupt 0

It's possibly acting weird because lack of the service coin in the definition, as the mcu is checking bits 5,6,7,
try adding:
    PORT_BIT( 0x1f, IP_ACTIVE_LOW, IPT_UNUSED )
    PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_SERVICE1 )

Reply:
and this:
m_subcpu->set_input_line(M6809_FIRQ_LINE, ASSERT_LINE); // FIRQ to main cpu
should be referring to the main cpu :)

Reply:
The unused bits should at least be pulled up.

Reply:
something like "PORT_BIT( 0x0f, IP_ACTIVE_LOW, IPT_UNUSED )"  ?
(the mcu masks p3 in with 0xf0 after reading)

Reply:
thanks guys :)

Reply:
> 
> 
> and this:
> m_subcpu->set_input_line(M6809_FIRQ_LINE, ASSERT_LINE); // FIRQ to main cpu
> should be referring to the main cpu :)

I think maybe the CPUs were swapped in MAME at some point (maybe to help with race conditions) because all other cases the comments says 'main' but uses 'sub'